job/cache.go

```diff
@@ -166,18 +166,20 @@ func (c *MemoryJobCache) PersistEvery(persistWaitTime time.Duration) {
 }
 
 type LockFreeJobCache struct {
-	jobs  *hashmap.HashMap
-	jobDB JobDB
+	jobs            *hashmap.HashMap
+	jobDB           JobDB
+	retentionPeriod time.Duration
 }
 
 func NewLockFreeJobCache(jobDB JobDB) *LockFreeJobCache {
 	return &LockFreeJobCache{
-		jobs:  hashmap.New(),
-		jobDB: jobDB,
+		jobs:            hashmap.New(),
+		jobDB:           jobDB,
+		retentionPeriod: -1,
 	}
 }
 
-func (c *LockFreeJobCache) Start(persistWaitTime time.Duration) {
+func (c *LockFreeJobCache) Start(persistWaitTime time.Duration, jobstatTtl time.Duration) {
```

Comment:
@QBui The tests need to be updated to incorporate the change in the signature of the `Start` function.